core/lib/include/assert.h

```diff
@@ -67,7 +67,7 @@ extern "C" {
 #endif /*__NORETURN*/
 
 #ifdef NDEBUG
-#define assert(ignore)((void)0)
+#define assert(cond) (cond)?((void)0):__builtin_unreachable()
```

Comment:
```suggestion
#define assert(cond) ((cond) ? (void)0 : __builtin_unreachable())
```
just to get this more in line with our other assert defines

while you are editing assert.h please remove the double space in line 117